jina/flow/base.py

```diff
@@ -712,6 +712,12 @@ def add(
             kwargs, parser, True, fallback_parsers=FALLBACK_PARSERS
         )
 
+        # Temporary workaround to re-import executor module when using mp spawn start
+        # method, so thta the executor is re-registered with pyyaml. A better solution
+        # probably involves deeper refactoring
+        if isinstance(kwargs.get('uses'), type(JAMLCompatible)):
+            args._exec_cls = kwargs['uses']
```

Comment:
Just trying to understand, I tried grepping for ` _exec_cls` but it seems like it's not used anywhere except here where it's being set and in  `jina/jaml/__init__.py` where it's deleted from `kwargs`. What does this `_exec_cls` do?

Reply:
Exactly, it's not used anywhere. 

It is only passed, because when you pass the class to `multiprocessing.Process.start` method (which is what happens under the hood when jina starts the pod/pea) when using `'spawn'` start method, it will re-import that class from its module. And this re-importing is the action that is needed to trigger it's registration with `pyyaml`.

Reply:
gotcha